src/module.d.ts

```diff
@@ -474,7 +489,7 @@ declare class posthog {
      *
      * @param {Function} [callback] The callback function will be called once the feature flags are ready. It'll return a list of feature flags enabled for the user.
      */
-    static onFeatureFlags(callback: (flags: string[]) => void): false | undefined
+    static onFeatureFlags(callback: (flags: string[] | Record<string, boolean | string>) => void): false | undefined
```

Comment:
Here's the thing: the type for `flags` depends upon the API version which is passed as a parameter to `decide()`. It's silly to expect a user of this library to pass a callback which could accept both types as they will likely only need to use one. `string[]` is only for backwards compat, in other words.